onnxruntime/test/optimizer/ensure_unique_dq_for_node_unit_test.cc

```diff
@@ -233,4 +233,24 @@ TEST(EnsureUniqueDQForNodeUnitTests, QDQWithMultiConsumerDQNodes) {
   EXPECT_EQ(OpCount(op_count_before, "DequantizeLinear") + 4, OpCount(op_count_after, "DequantizeLinear"));
 }
 
+TEST(EnsureUniqueDQForNodeUnitTests, QDQWithMultiConsumerDQNodesPreservingAttributes) {
+  constexpr auto model_uri = ORT_TSTR("testdata/qdq_with_multi_consumer_q_dq_axis.onnx");
+
+  SessionOptions session_options{};
+  // test interaction with level 1 transformers
+  session_options.graph_optimization_level = TransformerLevel::Level1;
+
+  InferenceSessionWrapper session{session_options, GetEnvironment()};
+
+  ASSERT_STATUS_OK(session.Load(model_uri));
+
+  const auto op_count_before = CountOpsInGraph(session.GetGraph());
+
+  ASSERT_STATUS_OK(session.Initialize());
+
+  const auto op_count_after = CountOpsInGraph(session.GetGraph());
+
+  EXPECT_EQ(OpCount(op_count_before, "DequantizeLinear") + 8, OpCount(op_count_after, "DequantizeLinear"));
```

Comment:
Hello @edgchen1 
Thanks for the suggestion. 
I tried to use the codes like following to fetch axis values of the DQ nodes in graph:
```
  for (auto& node : session.GetGraph().Nodes()) {
    if (node.OpType() == "DequantizeLinear") {
      auto attrs = node.GetAttributes();
      if (attrs.find("axis") != attrs.end()) {
        auto& axis_attr = attrs.at("axis");
        int64_t axis = axis_attr.i();
      }
    }
  }
}
```
But then I found all the fetched data is 1 in this test even before the session Initialize. This is quite strange, since in the input onnx model, the given axis is 0 for several DQ nodes, like:
![image](https://github.com/microsoft/onnxruntime/assets/19971688/b9aabd7c-23f9-4be0-a6ca-2c4f97a3a8be)

Do you have any idea what have been set wrong? Or, there might be another bug during onnxruntime graph construction?

Reply:
Hello @edgchen1 
Thanks for the suggestion. I have updated the test to include explicit attributes axis check for related nodes. Please review.